Frontend/lib/models/tournament.dart

```diff
@@ -0,0 +1,44 @@
+enum StartingMethod {
+  bulloff,
+  random,
+}
+
+class TournamentModel {
+  final String id;
+  final String name;
+  final String location;
+  final DateTime startTime;
+  final String? clubId;
+  final StartingMethod startingMethod;
+
+  TournamentModel({
+    required this.id,
+    required this.name,
+    required this.location,
+    required this.startTime,
+    this.clubId,
+    required this.startingMethod,
+  });
+
+  factory TournamentModel.fromJson(Map<String, dynamic> json) {
+    return TournamentModel(
+      id: json['id'],
+      name: json['name'],
+      location: json['location'],
+      startTime: DateTime.parse(json['start_time']),
+      clubId: json['club_id'],
+      startingMethod: StartingMethod.values.firstWhere((e) => e.toString().split('.').last == json['starting_method']),
+    );
+  }
+
+  Map<String, dynamic> toJson() {
+    return {
+      // 'id': id,
+      'name': name,
+      'location': location,
+      'start_time': startTime.toIso8601String(),
+      'club_id': clubId,
+      'starting_method': startingMethod.toString().split('.').last,
```

Comment:
Is `.split('.').last` nodig? Als je `toString()` doet dan krijg je toch al een string. Dat laatste is volgens mij overbodig.